### PR TITLE
change how extended network tests are selected/skipped

### PR DESCRIPTION
**Pull request by danwinship**
Currently the extended networking test requires you to set `NETWORKING_E2E_ISOLATION=true` when using the multitenant plugin, and `NETWORKING_E2E_NETWORKPOLICY=true` when using the networkpolicy plugin (or a third-party plugin). This changes it to just check the active network plugin at runtime, and select the right tests automatically based on that. Which means

1. When running the tests against an installed cluster with `OPENSHIFT_TEST_KUBECONFIG=... ./test/extended/networking.sh`, you don't have to also set the correct additional environment variables.
2. We can change conformance_gce to install multitenant rather than subnet, and the tests will adapt themselves correctly automatically.

I know at some point someone said we were supposed to decide what tests to do based on labels or whatever, but the checking-at-runtime approach seems to be more inline with, eg, `framework.SkipUnlessNodeCountIsAtLeast()`, `framework.SkipUnlessLocalEphemeralStorageEnabled()`, `framework.SkipUnlessNodeOSDistroIs()`, etc.

### CODE REVIEW

**Comment by danwinship** [comment]
Comments on this? This will let us switch conformance-gce over to the multitenant plugin, which will then make networking-minimal completely redundant, so we can get rid of it, which will get rid of docker-in-docker-related flakes like #13241.

File: test/extended/networking/util.go [hunk 1]
```diff
@@ -252,12 +254,42 @@ func checkConnectivityToHost(f *e2e.Framework, nodeName string, podName string,
...
+			pluginName = ""
+		}
+		cachedNetworkPluginName = &pluginName
+	}
+	return *cachedNetworkPluginName
+}
+
 func pluginIsolatesNamespaces() bool {
-	return os.Getenv("NETWORKING_E2E_ISOLATION") == "true"
+	if os.Getenv("NETWORKING_E2E_ISOLATION") == "true" {
```
[the remark this answers is not in the document]
Comment (danwinship) [reply test/extended/networking/util.go]: We actually have `test/extended/cni_vendor_test.sh` which is explicitly for third party plugin vendors to be able to run, to make sure that their plugin is "OpenShift-compatible". (In particular, even if they've tested against the upstream tests, this runs things like the router tests.)

Maybe we don't need to allow the overrides anyway (probably all third-party plugins do implement NetworkPolicy and don't implement ovs-multitenant-style isolation). But it was just a few lines of code so I left it there...

[the remark this answers is not in the document]
Comment (smarterclayton) [reply test/extended/networking/util.go]: OK, I can live with it for now.

**Comment by danwinship** [comment]
/retest
(extended_networking_minimal failure is #18765, an unrelated docker-in-docker flake)

**Pull request merged**